### PR TITLE
Updating New Release Info topic

### PR DESCRIPTION
**Pull request by pstack2021**
## Why

The information was split between multiple pages and was also kind of redundant with the release notes on GitHub.

It was time to clarify and make the content more readable.

## How

The `changelog` and `news` pages are gone. They didn't bring much values compared to what we had on GitHub. They barely served as announcements. With RabbitMQ 3.13.0, we started to use the blog to publish announcements, so they are not needed anymore.

The `versions` page was cleaned up. It now show the release branches and their state of support in the same way as [Linux distribution versions on Wikipedia](https://en.wikipedia.org/wiki/Debian_version_history). It is now possible for a user to see what is supported at a glance.

The really old releases were deleted from the page. We don't need to know that RabbitMQ 3.1.x is no longer supported.

The `versions.md` page was renamed to `index.md`. Thus the new URL is simply `/release-information`.

All release details are put in `docusaurus.config.js`. This was already used to know what the latest patch was on a per-release-branch basis. Now, we also set the release and end-of-support dates. The component selects the colors based on the compile-time date and the end-of-support dates. Therefore, the colors will be automatically updated with each build of the website. This is not rendered at runtime, so if there are no builds of the website, the color will be outdated. I believe that's ok because support expires at the same time a new release series is published.

They are rendered as a table using a React/MDX component in `src/components/RabbitMQServerReleaseInfo`.

### CODE REVIEW

**Comment by pstack2021** [comment]
Hi @dumbbell,

Can you please review this new topic, many thanks. Note, the links are to live html files for now, I know they have to change.

Two additional questions: 

1. I have the base branch now changed to: `paula_toc_updates_for_new_site` which I assume its correct? the `release_information` branch will be merged into `paula_toc_updates_for_new_site` and then `paula_toc_updates_for_new_site` will be merged into `main`?
2. The drop down menu on the site has a "Release Information" option, which opens: https://www.rabbitmq.com/docs/versions should that point to this topic now instead? 

Many thanks in advance.
Paula.

**Comment by mkuratczyk** [comment]
I'd merge "Currently Supported Versions" into the "Release Information" (main page of this section). Right now after clicking on the Release Information in the main docs, you are taken to a page which has very little information and is somewhat misleading since it mentions 3.11 as supported, which is not really true for most users. Displaying the more detail "Currently Supported Versions" page would provide more accurate information faster.

`Project Announcements` has a very exciting title for what is currently a very boring page, which makes for a disappointing experience. I'm not sure if there's a point in having this page until we have more exciting announcements there.
If we keep it, then there's a problem with how dates are displayed (too close to the title in my opinion):
Screenshot 2024-03-08 at 08 39 10

**Comment by dumbbell** [comment]
>     1. I have the base branch now changed to: `paula_toc_updates_for_new_site` which I assume its correct? the `release_information` branch will be merged into `paula_toc_updates_for_new_site` and then `paula_toc_updates_for_new_site` will be merged into `main`?

`paula_toc_updates_for_new_site` will be merged to `main` when it’s ready. In parallel, `release_info` will be "rebased" on to `paula_toc_updates_for_new_site` until it’s merged, then on to `main`. Once `release_info` is ready, it will be merged too.

>     2. The drop down menu on the site has a "Release Information" option, which opens: https://www.rabbitmq.com/docs/versions should that point to this topic now instead?

Yes problably.

Warning: you created this pull request using `paula_toc_updates_for_new_site` instead of `release_info`. Therefore, we can’t view your work.

**Comment by dumbbell** [comment]
I agree with @mkuratczyk: we could have the `versions.md` page renamed to `index.md` and simplified a bit. The table could point to announcements on the blog and release notes in the versioned docs.

We will surely start to produce errata notes as well to list known problems in a release. They would live in the versioned docs as well like release notes and would be linked in that table.

The "In service" column can go away, it’s useless.

Here is an example from FreeBSD: https://www.freebsd.org/releases/
(don’t pay attention to the old school design :)

`news.md` and `changelog.md` are just garbage at this poing IMHO. I would just removed them.

**Comment by pstack2021** [comment]
Thanks very much @mkuratczyk and @dumbbell for your feedback.

**I have implemented the following changes:**
 - Renamed versions.md to index.md
 - Renamed the "Latest Patch" column to "Latest Patch/Link to Release Notes" because that is what the link in those rows is now anyway, lets make it clearer.
 - I removed the changelog.md and news.md files. 
 - I updated the introductory paragraph. I got rid of references to "release series" and now use the word "releases" instead. 
 - I removed the "In Service For" column.
 - I added a paragraph, which includes a link to the blogs page on the site. I did not include a column in the tables for this mainly because we don't have a TOC for the blogs i.e. if we had a TOC entry "3.13 blogs", then in the 3.13 row in the table, I could add a link to "3.13 blogs", I did consider including links to every 3.13 blog in the 3.13 row in the table but that would need constant maintenance, for example, if more 3.13 blogs were added to the blog page in the future then the rows in the tables here would need to be updated to include them also).
 
 **Outstanding questions:**
1. @mkuratczyk @dumbbell: Is this information **For guidance on upgrades, see the [Upgrade](https://www.rabbitmq.com/docs/upgrade) and [Blue/Green Deployment Upgrade](https://www.rabbitmq.com/docs/blue-green-upgrade) guides.** still useful? will I leave it in? I guess if users are on an X release and we want them to upgrade to the next release then yes if it useful if the docs linked here will successfully upgrade them :) I could reword it to "**_If you want to upgrade from one release to another, use the_ [Upgrade](https://www.rabbitmq.com/docs/upgrade) and [Blue/Green Deployment Upgrade](https://www.rabbitmq.com/docs/blue-green-upgrade) _documentation._** 

2. @dumbbell what do you mean by **The table could point to announcements on the blog and release notes in the versioned docs.** above? what release notes in versioned docs? are you referring to doing this in the future i.e. having the release notes in the versioned docs on this site (which we discussed a long time ago), for now the RabbitMQ release notes are on Github.

**Comment by pstack2021** [comment]
@dumbbell will you update the `docusaurus.config.mjs` file please to change:

```
 {
              "to": "/release-information/versions",
              "label": "Release information"
            }
```

to:

```
 {
              "to": "/release-information",
              "label": "Release Information"
            }

```
so that the "Release information" drop down menu now points to the index.md file in the release-information folder. I was going to do it myself but then saw in bold "DON'T EDIT THIS FOLDER", many thanks.

**Comment by dumbbell** [comment]
>     * Renamed the "Latest Patch" column to "Latest Patch/Link to Release Notes" because that is what the link in those rows is now anyway, lets make it clearer.

I preferred the previous title: it was shorter and I’m not sure we need to tell what the link points to. We are on the Release information page, thus we can assume that the links on a version will point to details about the release.

>     * I removed the changelog.md and news.md files. The commit will remove them completely, do we want to do this? Do we need a copy of them to go through them incase there is anything we want to pull from them when we are doing further content optimisation in the future i.e. for 4.0? I guess they are available in the history.

Removing them is the right thing IMHO. The files didn’t contain anything useful.

>     * I added a paragraph, which includes a link to the blogs page on the site. I did not include a column in the tables for this mainly because we don't have a TOC for the blogs i.e. if we had TOC entries, lets say we had a "3.13 blogs" TOC entry, then in the 3.13 row in the table, I could add a link to "3.13 blogs". I did consider including links to every 3.13 blog in the 3.13 row in the table but that would need constant maintenance, for example, if more 3.13 blogs were added to the blog page in the future then the rows in the tables here would need to be updated to include them also.

I agree, links to the blog posts (except the upcoming announcement one) are not needed. The introduction paragraph could point to the "Announcement" tag to filter blog posts that talk about releases: http://localhost:3000/blog/tags/announcements (though the tag needs some curation).

> **Outstanding questions:**
> 
>     1. @mkuratczyk @dumbbell: Is this information **For guidance on upgrades, see the [Upgrade](https://www.rabbitmq.com/docs/upgrade) and [Blue/Green Deployment Upgrade](https://www.rabbitmq.com/docs/blue-green-upgrade) guides.** still useful? will I leave it in? I guess if users are on an X release and we want them to upgrade to the next release then yes if it useful if the docs linked here will successfully upgrade them :) I could reword it to "**_If you want to upgrade from one release to another, use the_ [Upgrade](https://www.rabbitmq.com/docs/upgrade) and [Blue/Green Deployment Upgrade](https://www.rabbitmq.com/docs/blue-green-upgrade) _documentation._**

We can keep the paragraph, it is useful, even if release notes may give more version-specific details.

>     3. @dumbbell what do you mean by **The table could point to announcements on the blog and release notes in the versioned docs.** above? what release notes in versioned docs? are you referring to doing this in the future i.e. having the release notes in the versioned docs on this site (which we discussed a long time ago), for now the RabbitMQ release notes are on Github.

That would be in the future if we start to write release notes on the website.

>     4. @dumbbell for some reason the mpl.md file is the release_information folder, not sure why, I can remove it, it is not added to the sidebarsReleaseInfo.js file yet so hence not displaying in the "Release Information" TOC.

Don’t worry about this `mpl.md` file, it is gone in #1795.

**Review by dumbbell — APPROVED** [review]
